### DISCUSSION TITLE
Implict Fucntions declaration

### DISCUSSION THREAD

**Discussion by M-Rizwan-Anwar** [espressif/arduino-esp32#9283] [Q&A]
Dear community members, I am using arduino as a component in VS Code, I am having issues in ESP32 Core header files. it is saying implicit declaration for some of the i2c functions:

D:/esp32/Test-Helloworld/hello_world/components/cores/esp32/esp32-hal-i2c-slave.c:517:5: error: implicit declaration of function 'i2c_ll_cal_bus_clk' [-Werror=implicit-function-declaration]
  517 |     i2c_ll_cal_bus_clk(APB_CLK_FREQ, clk_speed, &clk_cal);
      |     ^~~~~~~~~~~~~~~~~~
D:/esp32/Test-Helloworld/hello_world/components/cores/esp32/esp32-hal-i2c-slave.c:525:5: error: implicit declaration of function 'i2c_ll_set_bus_timing'; did you mean 'i2c_ll_set_scl_timing'? [-Werror=implicit-function-declaration]
  525 |     i2c_ll_set_bus_timing(i2c->dev, &clk_cal);
      |     ^~~~~~~~~~~~~~~~~~~~~
      |     i2c_ll_set_scl_timing
D:/esp32/Test-Helloworld/hello_world/components/cores/esp32/esp32-hal-i2c-slave.c:526:5: error: implicit declaration of function 'i2c_ll_set_filter'; did you mean 'i2c_ll_set_tout'? [-Werror=implicit-function-declaration]
  526 |     i2c_ll_set_filter(i2c->dev, 3);
      |     ^~~~~~~~~~~~~~~~~
      |     i2c_ll_set_tout
      
      
      I tried multiple solutions, but nothing worked for me. Can anyone help me solve this error?

**Comment by M-Rizwan-Anwar** [comment]
I am having same issue with this also:
D:/esp32/Test-Helloworld/hello_world/components/cores/esp32/Client.h:38:17: error: 'virtual int Client::connect(const char*, uint16_t)' was hidden [-Werror=overloaded-virtual=]
   38 |     virtual int connect(const char *host, uint16_t port) =0;
      |                 ^~~~~~~

**Comment by SuGlider** [comment]
This seems already covered in other open discussion. Closing it.

**Comment by gfr10598** [comment]
Can you add a link to the open discussion where this is being (has been) covered?
